configs/named_values.go

```diff
@@ -250,6 +267,152 @@ func (m VariableParsingMode) Parse(name, value string) (cty.Value, hcl.Diagnosti
 	}
 }
 
+// VariableValidation represents a configuration-defined validation rule
+// for a particular input variable, given as a "validation" block inside
+// a "variable" block.
+type VariableValidation struct {
+	// Condition is an expression that refers to the variable being tested
+	// and contains no other references. The expression must return true
+	// to indicate that the value is valid or false to indicate that it is
+	// invalid. If the expression produces an error, that's considered a bug
+	// in the module defining the validation rule, not an error in the caller.
+	Condition hcl.Expression
+
+	// ErrorMessage is one or more full sentences, usually in English for
```

Comment:
> usually in English

I'm going to nitpick here, but my more recent experience writing terraform code was for a team that I'm _confident_ will write error messages in Dutch.

There's no guarantee that a user will want to write their error messages in english. It might be more accurate to indicate that it's assumed to be english and `looksLikeSentences` will complain if it doesn't like the punctuation.

I feel rather (oddly) strongly about this, even though it's a tiny detail, but not at all in a blocking way if you disagree (super helpful, I'm sure!).

Reply:
The "usually in English" here is coming from the fact that a message written in Dutch would end up looking like this in the UI (though presumbly written in correct Dutch, not this Google Translate mess for the sake of example):

```
Error: Invalid value for variable

  on example.tf line 5, in module "example":
   5:   ami_id = "amo-abc123"

Het argument "ami_id" moet een geldige AMI-ID zijn, beginnend met "ami-".

This was checked by the validation rule at modules/example/variables.tf:7,3-13.
```

I'm guessing that as long as Terraform's own UI is only available in English the UI of most modules will be in English too. There's no code here preventing you from writing messages in other languages of course, if the author is happy to see some non-English text mixed in with predominately-English. The "usually" is an assumption about how it will be commonly used, not a usage requirement, and I wrote it here specifically to _acknowledge_ that our i18n story for this feature (and Terraform as a whole) is poor.